src/catalog/tests/open.rs

```diff
@@ -215,17 +216,26 @@ async fn open_check<D: DurableCatalogState>(
 
         // Initialize the stash.
         {
-            let _ = openable_state
+            let mut state = openable_state
                 .open(SYSTEM_TIME.clone(), &bootstrap_args(), None)
                 .await
                 .unwrap();
+            assert_eq!(
+                state.epoch(),
+                NonZeroI64::new(2).expect("known to be non-zero")
+            );
         }
 
         // Open catalog in check mode.
         let mut state = openable_state
             .open_savepoint(SYSTEM_TIME.clone(), &bootstrap_args(), None)
             .await
             .unwrap();
+        // Savepoint catalogs do not increment the epoch.
+        assert_eq!(
+            state.epoch(),
+            NonZeroI64::new(2).expect("known to be non-zero")
+        );
```

Comment:
This seems somewhat wrong to me, but would require a bit of a refactor in the stash crate to fix.

Reply:
You think savepoints should increment the epoch?

Reply:
From the outside savepoint behave mostly as a writable stash. All other writes look like they actually happen, so intuitively I thought that would happen for the epoch too. As in the epoch would get incremented in a savepoint transaction but never committed. Though I guess this isn't the case for creating the tables in savepoint mode.

Another thing I was thinking about is that the savepoint mode is meant to test an upgrade. In an upgrade the epoch will get incremented so it's somewhat possible that the upgrade test doesn't actually mirror what would happen in an upgrade.

Reply:
Compelling! I like that line of thinking a lot.

Reply:
https://github.com/MaterializeInc/materialize/issues/22378